Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -135,29 +135,33 @@ ISymbolTableRelocator symbolTableRelocator
       }
     }
 
-    public Task<DafnyDocument> SaveDocumentAsync(TextDocumentIdentifier documentId) {
+    public async Task<DafnyDocument> SaveDocumentAsync(TextDocumentIdentifier documentId) {
```

Comment:
Why change this method to async?

Reply:
This change is to that it's possible to `await` the result of `VerifyDocumentIfRequiredAsync` as @camrein suggested. Given that I also moved the exception handler into `VerifyDocumentIfRequiredAsync`, it may not strictly be required, but it also makes it more closely match the structure of `UpdateDocumentAsync`, which seems good for making the code easier to understand.

Reply:
I agree it might be more readable 👍